src/libasr/codegen/wasm_to_x64.cpp

```diff
@@ -35,14 +35,17 @@ class X64Visitor : public WASMDecoder<X64Visitor>,
     int32_t last_vis_i32_const, last_last_vis_i32_const;
     std::map<std::string, std::string> label_to_str;
     std::vector<std::string> if_unique_id;
+    std::map<std::string, double> double_consts;
 
+    bool decoding_data_segment;
```

Comment:
Can you please add a comment here what this does (I assume it is true if we are decoding a data segment in WASM, and I can also see you are sometimes only executing some logic if not decoding a segment, but I do not understand why).

Reply:
A data segment in `wasm` uses a set of instructions/expressions to specify the offset in memory where the string/data is to be stored.

For example, in `(data (;0;) (i32.const 4) "abc")`, the set of instructions/expressions contains one element which is `i32.const 4` and it evaluates to `4` which represents the memory offset where the string `abc` is to be stored. 

To obtain the string offset we need to decode these set of instructions. Since, we currently support compile-time strings and the string offset is stored in `last_vis_i32_const`, the instructions are not needed for us at the moment. Since, they are not needed, we should not be emitting them while decoding a data segment.

Reply:
I added a comment about it in commit [ASM_X64: Add comment about decoding_data_segment](https://github.com/lcompilers/lpython/pull/1402/commits/16217d7a8e07f01ce687faef21c567b045402444).